applications/HDF5Application/custom_utilities/container_io_utils.h

```diff
@@ -0,0 +1,477 @@
+        return "NONHISTORICAL";
+    } else if constexpr(std::is_same_v<TContainerIOType, FlagIO>) {
+        return "FLAGS";
+    } else if constexpr(std::is_same_v<TContainerIOType, BossakIO>) {
+        return "HISTORICAL_BOSSAK";
+    } else if constexpr(std::is_same_v<TContainerIOType, VertexValueIO>) {
+        return "INTERPOLATED";
+    } else if constexpr(std::is_same_v<TContainerIOType, GaussPointValueIO>) {
+        return "GAUSS_POINT";
+    } else {
+        static_assert(!std::is_same_v<TContainerIOType, TContainerIOType>, "Unsupported container io type.");
+    }
+}
+
+template<class TContainerType>
+std::string GetContainerType()
+{
+    if constexpr(std::is_same_v<TContainerType, ModelPart::NodesContainerType>) {
+        return "NODES";
+    } else if constexpr(std::is_same_v<TContainerType, ModelPart::ConditionsContainerType>) {
+        return "CONDITIONS";
+    } else if constexpr(std::is_same_v<TContainerType, ModelPart::ElementsContainerType>) {
+        return "ELEMENTS";
+    } else if constexpr(std::is_same_v<TContainerType, Detail::VertexContainerType>) {
+        return "VERTICES";
+    } else {
+        static_assert(!std::is_same_v<TContainerType, TContainerType>, "Unsupported container type.");
+    }
+}
+
+template<class TContainerType>
+auto& GetLocalContainer(ModelPart& rModelPart)
+{
+    if constexpr(std::is_same_v<TContainerType, ModelPart::NodesContainerType>) {
+        return rModelPart.GetCommunicator().LocalMesh().Nodes();
+    } else if constexpr(std::is_same_v<TContainerType, ModelPart::ConditionsContainerType>) {
+        return rModelPart.GetCommunicator().LocalMesh().Conditions();
+    } else if constexpr(std::is_same_v<TContainerType, ModelPart::ElementsContainerType>) {
+        return rModelPart.GetCommunicator().LocalMesh().Elements();
+    } else {
+        static_assert(!std::is_same_v<TContainerType, TContainerType>, "Unsupported container type.");
+    }
+}
+
+template<class TContainerType>
+const auto& GetLocalContainer(const ModelPart& rModelPart)
+{
+    if constexpr(std::is_same_v<TContainerType, ModelPart::NodesContainerType>) {
+        return rModelPart.GetCommunicator().LocalMesh().Nodes();
+    } else if constexpr(std::is_same_v<TContainerType, ModelPart::ConditionsContainerType>) {
+        return rModelPart.GetCommunicator().LocalMesh().Conditions();
+    } else if constexpr(std::is_same_v<TContainerType, ModelPart::ElementsContainerType>) {
+        return rModelPart.GetCommunicator().LocalMesh().Elements();
+    } else {
+        static_assert(!std::is_same_v<TContainerType, TContainerType>, "Unsupported container type.");
+    }
+}
+
+template<class TContainerType, class TComponentType, class TContainerDataIO, class TDataType>
+void CopyToContiguousDataArray(
+    const TContainerType& rContainer,
+    const TComponentType& rComponent,
+    const TContainerDataIO& rContainerDataIO,
+    TDataType* pBegin)
```

Comment:
can you please also pass the array's size as well? It'll make life much easier if we could validate that the array is big enough to fit all entries, instead of segfaulting.

```suggestion
    const TContainerType& rContainer,
    const TComponentType& rComponent,
    const TContainerDataIO& rContainerDataIO,
    TDataType* pBegin,
    const std::size_t Size)
```

If this pattern occurs frequently, we could also add a `Span` (similar to [`std::span`](https://en.cppreference.com/w/cpp/container/span)) class that models a view over a contiguous array. It's a handy, safe and lightweight thing to pass around compared to just juggling with raw pointers.

Reply:
this one slipped under your radar

Reply:
done :)